spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/it/SecretManagerIntegrationTests.java

```diff
@@ -202,19 +185,21 @@ private void createSecretPayload(String data) {
 		client.addSecretVersion(payloadRequest);
 	}
 
-	private boolean secretExists(String secretId, String version) {
-		String projectId = projectIdProvider.getProjectId();
-		ProjectName projectName = ProjectName.of(projectId);
+	private boolean secretExists(String secretId) {
+		ProjectName projectName = ProjectName.of(projectIdProvider.getProjectId());
 		ListSecretsPagedResponse listSecretsResponse = this.client.listSecrets(projectName);
 		return StreamSupport.stream(listSecretsResponse.iterateAll().spliterator(), false)
-				.filter(secret -> secret.getName().contains(secretId))
-				.anyMatch(secret -> {
-					SecretVersionName secretVersionName = SecretVersionName.newBuilder()
-							.setProject(projectId)
-							.setSecret(secretId)
-							.setSecretVersion(version)
-							.build();
-					return this.client.accessSecretVersion(secretVersionName) != null;
-				});
+				.anyMatch(secret -> secret.getName().contains(secretId));
+	}
+
+	private void deleteSecret(String secretId) {
+		try {
+			this.client.deleteSecret(SecretName.of(this.projectIdProvider.getProjectId(), secretId));
+		}
+		catch (Exception e) {
```

Comment:
Again, why not just catch `NotFoundException`?

Reply:
Done.